code/controllers/subsystems/jobs.dm

```diff
@@ -105,8 +124,10 @@ SUBSYSTEM_DEF(jobs)
 				must_fill_titles += job.title
 			if(job.department_refs)
 				for(var/dept_ref in job.department_refs)
-					if(dept_ref in SSdepartments.departments)
-						LAZYDISTINCTADD(positions_by_department[dept_ref], job.title)
+					var/decl/department/dept = SSjobs.get_department_by_reference(dept_ref)
+					if(dept)
+						LAZYDISTINCTADD(positions_by_department[dept.type], job.title)
+						LAZYDISTINCTADD(positions_by_department[dept.name], job.title)
```

Comment:
Doesn't need to be a thing anymore. Seemed to be mostly used by jobbans, and those use department types now.
Only place left doing this is `event_dynamic.dm` and they should be converted to using types too